### PR TITLE
Add functionality to write fasta with protein sequences

### PR DESCRIPTION
**Pull request by christopher-mohr**
Add functionality to write out a FASTA file with protein sequences as mentioned in #9 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Review by skrakau — CHANGES_REQUESTED** [review]
Hi, great that you added this! It is addressing https://github.com/nf-core/epitopeprediction/issues/9 (not https://github.com/nf-core/epitopeprediction/pull/33, although both dealing with FASTAs). I added a few comments.

Could you additionally add a description to the `docs/output.md`? 

And if I see it right, this option only works (and makes sense) with variants as input, i.e. not with `--peptides` or `--proteins`. Maybe it would be useful to check at the beginning of the pipline if the parameter combination is valid and throw an error if not. Also this could be mentioned in the help/schema description of this parameter.

**Comment by christopher-mohr** [comment]
> Hi, great that you added this! It is addressing #9 (not #33, although both dealing with FASTAs). I added a few comments.

Right, changed it! 👍

**Comment by christopher-mohr** [comment]
Thanks for the review @skrakau! Hope you're happy with the changes :)

File: docs/output.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ HTHVYLFLS 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpe
 HVYLFLSNL 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpeithi-1.0 0.0 0.0 False False False c.173C>A p.Pro58His
 ```
 
+When the parameter `--fasta_output` is specified a `FASTA` file will be generated that contains the sequences of proteins that are affected by the provided genomic variants. The resulting `FASTA` file will contain the wild-type and mutated protein sequences.
```
Comment (skrakau) [inline docs/output.md:36]: In which folder will this be located, and how will it be called?

**Review by skrakau — CHANGES_REQUESTED** [review]
Looks good, just one tiny detail is still missing ;)

**Comment by christopher-mohr** [comment]
The last commit includes more changes than initially planned ;):

- merge FASTA files in case of splitted results 
- fixed the statistics for single and multiple (splitter) results
- include input base name in result files
- change result folder name
- update docs

File: docs/output.md [hunk 2]
```diff
@@ -16,11 +16,11 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
 
 [FRED2](https://github.com/FRED-2) is used to perform the prediction of Epitopes on the given data, independent of the chosen `tools` to perform the prediction.
 
-**Output directory: `results/`**
+**Output directory: `predictions/`**
 
-* `prediction_report.json`
+* `${input_base_name}_prediction_report.json`
```
Comment (skrakau) [inline docs/output.md:21]: This is inconsistent with the notation used for `[tool].[version].supported_alleles.txt`. Seems every pipeline does this differently. Use either `[...]` or `${...}` in the whole `output.md`.

File: docs/output.md [hunk 3]
```diff
@@ -16,11 +16,11 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
...
   * The predicted epitopes in JSON format for downstream analysis tasks
-* `prediction_report.tsv`
+* `${input_base_name}_prediction_results.tsv`
```
Comment (skrakau) [inline docs/output.md:23]: See above.

File: docs/output.md [hunk 4]
```diff
@@ -33,6 +33,13 @@ HTHVYLFLS 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpe
 HVYLFLSNL 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpeithi-1.0 0.0 0.0 False False False c.173C>A p.Pro58His
 ```
 
+When the parameter `--fasta_output` is specified a `FASTA` file will be generated that contains the sequences of proteins that are affected by the provided genomic variants. The resulting `FASTA` file will contain the wild-type and mutated protein sequences.
+
+**Output directory: `predictions/`**
+
+* `${input_base_name}_prediction_proteins.fasta`
```
Comment (skrakau) [inline docs/output.md:40]: See above.

**Review by skrakau — CHANGES_REQUESTED** [review]
Looks good! :) 

Just one remaining inconsistency in the notation of output files.

**Review by skrakau — APPROVED** [review]
LGTM @christopher-mohr

**Comment by christopher-mohr** [comment]
> LGTM @christopher-mohr

Thank you @skrakau !

**Pull request merged**